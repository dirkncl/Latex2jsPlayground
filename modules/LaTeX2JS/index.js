export * from './latex2html5/index.js'