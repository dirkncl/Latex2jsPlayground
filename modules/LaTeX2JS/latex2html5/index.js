//import latex2js_css from '../latex2js/src/lib/latex2js.css' with {type: "css"}
//import latex2js_mathapedia_css from '../latex2js/src/lib/latex2js.mathapedia.css' with {type: "css"}
//export {latex2js_css, latex2js_mathapedia_css}

export * from './src/index.js'