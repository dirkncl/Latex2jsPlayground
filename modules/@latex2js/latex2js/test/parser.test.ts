import LaTeX2JS from '../src/src/index.js';
const latex = new LaTeX2JS();

describe('Parser', () => {
  it('parser', () => {
    const parser = latex.getParser();
    const parsed = parser.parse(`
Let's get to the point. The core of PSTricks is graphics!

\\begin{center}
\\begin{pspicture}(-5,-5)(5,5)
\\psline{->}(0,-3.75)(0,3.75)
\\psline{->}(-3.75,0)(3.75,0)
\\pscircle(0,0){ 3 }
\\end{pspicture}
\\end{center}

which can be produced using the following \\TeX:

\\begin{verbatim}
\\begin{center}
\\begin{pspicture}(-5,-5)(5,5)
\\psline{->}(0,-3.75)(0,3.75)
\\psline{->}(-3.75,0)(3.75,0)
\\pscircle(0,0){ 3 }
\\end{pspicture}
\\end{center}
\\end{verbatim}
    `);
    expect(parsed).toMatchSnapshot();
  });
  it('parse', () => {
    const result = latex.parse(`
Let's get to the point. The core of PSTricks is graphics!

\\begin{center}
\\begin{pspicture}(-5,-5)(5,5)
\\psline{->}(0,-3.75)(0,3.75)
\\psline{->}(-3.75,0)(3.75,0)
\\pscircle(0,0){ 3 }
\\end{pspicture}
\\end{center}

which can be produced using the following \\TeX:

\\begin{verbatim}
\\begin{center}
\\begin{pspicture}(-5,-5)(5,5)
\\psline{->}(0,-3.75)(0,3.75)
\\psline{->}(-3.75,0)(3.75,0)
\\pscircle(0,0){ 3 }
\\end{pspicture}
\\end{center}
\\end{verbatim}
    `);
    expect(result).toMatchSnapshot();
  });
});
