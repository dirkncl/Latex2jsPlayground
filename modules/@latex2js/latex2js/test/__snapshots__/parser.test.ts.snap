// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser parse 1`] = `
[
  {
    "lines": [
      "<br>",
      "Let's get to the point. The core of PSTricks is graphics!",
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "xunit": 50,
      "yunit": 50,
    },
    "type": "math",
  },
  {
    "env": {
      "h": 10,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "lines": [
      "\\psline{->}(0,-3.75)(0,3.75)",
      "\\psline{->}(-3.75,0)(3.75,0)",
      "\\pscircle(0,0){ 3 }",
    ],
    "match": [
      "\\begin{pspicture}(-5,-5)(5,5)",
      "-5",
      "-5",
      "5",
      "5",
    ],
    "plot": {
      "psarc": [],
      "psaxes": [],
      "pscircle": [
        {
          "data": {
            "cx": 250,
            "cy": 250,
            "r": 150,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\pscircle(0,0){ 3 }",
            "0",
            "0",
            " 3 ",
          ],
        },
      ],
      "psframe": [],
      "psline": [
        {
          "data": {
            "arrows": [
              0,
              1,
            ],
            "dots": [
              0,
              0,
            ],
            "fillcolor": "black",
            "fillstyle": "solid",
            "linecolor": "black",
            "linestyle": "solid",
            "linewidth": 2,
            "x1": 250,
            "x2": 250,
            "y1": 437.5,
            "y2": 62.5,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\psline{->}(0,-3.75)(0,3.75)",
            undefined,
            "{->}",
            "0",
            "-3.75",
            "(0,3.75)",
            "0",
            "3.75",
          ],
        },
        {
          "data": {
            "arrows": [
              0,
              1,
            ],
            "dots": [
              0,
              0,
            ],
            "fillcolor": "black",
            "fillstyle": "solid",
            "linecolor": "black",
            "linestyle": "solid",
            "linewidth": 2,
            "x1": 62.5,
            "x2": 437.5,
            "y1": 250,
            "y2": 250,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\psline{->}(-3.75,0)(3.75,0)",
            undefined,
            "{->}",
            "-3.75",
            "0",
            "(3.75,0)",
            "3.75",
            "0",
          ],
        },
      ],
      "pspicture": [],
      "psplot": [],
      "pspolygon": [],
      "psset": [],
      "rput": [],
      "slider": [],
      "userline": [],
      "uservariable": [],
    },
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "pspicture",
  },
  {
    "lines": [
      "<br>",
      "which can be produced using the following $\\TeX$:",
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "math",
  },
  {
    "lines": [
      "\\begin{pspicture}(-5,-5)(5,5)",
      "\\psline{->}(0,-3.75)(0,3.75)",
      "\\psline{->}(-3.75,0)(3.75,0)",
      "\\pscircle(0,0){ 3 }",
      "\\end{pspicture}",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "verbatim",
  },
  {
    "lines": [
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "math",
  },
]
`;

exports[`Parser parser 1`] = `
[
  {
    "lines": [
      "<br>",
      "Let's get to the point. The core of PSTricks is graphics!",
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "xunit": 50,
      "yunit": 50,
    },
    "type": "math",
  },
  {
    "env": {
      "h": 10,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "lines": [
      "\\psline{->}(0,-3.75)(0,3.75)",
      "\\psline{->}(-3.75,0)(3.75,0)",
      "\\pscircle(0,0){ 3 }",
    ],
    "match": [
      "\\begin{pspicture}(-5,-5)(5,5)",
      "-5",
      "-5",
      "5",
      "5",
    ],
    "plot": {
      "psarc": [],
      "psaxes": [],
      "pscircle": [
        {
          "data": {
            "cx": 250,
            "cy": 250,
            "r": 150,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\pscircle(0,0){ 3 }",
            "0",
            "0",
            " 3 ",
          ],
        },
      ],
      "psframe": [],
      "psline": [
        {
          "data": {
            "arrows": [
              0,
              1,
            ],
            "dots": [
              0,
              0,
            ],
            "fillcolor": "black",
            "fillstyle": "solid",
            "linecolor": "black",
            "linestyle": "solid",
            "linewidth": 2,
            "x1": 250,
            "x2": 250,
            "y1": 437.5,
            "y2": 62.5,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\psline{->}(0,-3.75)(0,3.75)",
            undefined,
            "{->}",
            "0",
            "-3.75",
            "(0,3.75)",
            "0",
            "3.75",
          ],
        },
        {
          "data": {
            "arrows": [
              0,
              1,
            ],
            "dots": [
              0,
              0,
            ],
            "fillcolor": "black",
            "fillstyle": "solid",
            "linecolor": "black",
            "linestyle": "solid",
            "linewidth": 2,
            "x1": 62.5,
            "x2": 437.5,
            "y1": 250,
            "y2": 250,
          },
          "env": {
            "h": 10,
            "w": 10,
            "x0": -5,
            "x1": 5,
            "xunit": 50,
            "y0": -5,
            "y1": 5,
            "yunit": 50,
          },
          "fn": [Function],
          "match": [
            "\\psline{->}(-3.75,0)(3.75,0)",
            undefined,
            "{->}",
            "-3.75",
            "0",
            "(3.75,0)",
            "3.75",
            "0",
          ],
        },
      ],
      "pspicture": [],
      "psplot": [],
      "pspolygon": [],
      "psset": [],
      "rput": [],
      "slider": [],
      "userline": [],
      "uservariable": [],
    },
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "pspicture",
  },
  {
    "lines": [
      "<br>",
      "which can be produced using the following $\\TeX$:",
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "math",
  },
  {
    "lines": [
      "\\begin{pspicture}(-5,-5)(5,5)",
      "\\psline{->}(0,-3.75)(0,3.75)",
      "\\psline{->}(-3.75,0)(3.75,0)",
      "\\pscircle(0,0){ 3 }",
      "\\end{pspicture}",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "verbatim",
  },
  {
    "lines": [
      "<br>",
    ],
    "settings": {
      "fillstyle": "none",
      "h": 10,
      "linecolor": "black",
      "linestyle": "solid",
      "runit": 50,
      "unit": 50,
      "w": 10,
      "x0": -5,
      "x1": 5,
      "xunit": 50,
      "y0": -5,
      "y1": 5,
      "yunit": 50,
    },
    "type": "math",
  },
]
`;
